aidial_adapter_openai/gpt4_multi_modal/transformation.py

```diff
@@ -0,0 +1,129 @@
+from typing import List, Optional, cast
+
+from pydantic import BaseModel, root_validator
+
+from aidial_adapter_openai.gpt4_multi_modal.download import (
+    ImageFail,
+    download_image,
+)
+from aidial_adapter_openai.gpt4_multi_modal.image_tokenizer import (
+    tokenize_image,
+)
+from aidial_adapter_openai.gpt4_multi_modal.messages import (
+    create_image_message,
+    create_text_message,
+)
+from aidial_adapter_openai.utils.image_data_url import ImageDataURL
+from aidial_adapter_openai.utils.log_config import logger
+from aidial_adapter_openai.utils.storage import FileStorage
+from aidial_adapter_openai.utils.tokens import Tokenizer
+
+
+class MessageTransformResult(BaseModel):
```

Comment:
Why not split into two types:

`MessageTransformResult = MessageTransformError | MessageTransformSuccess`

Reply:
https://github.com/epam/ai-dial-adapter-openai/pull/150#discussion_r1763345034